_posts/2020-08-11-relay-pagingation-containers-demystified.markdown

```diff
@@ -0,0 +1,234 @@
+---
+layout: epic
+title: "Relay Pagination Containers Demystified"
+date: 2020-08-11
+categories: [react, relay, culture]
+author: ash
+---
+
+In 2017, Artsy adopted [Relay](https://relay.dev/en/) in both its front-end web and iOS codebases (using React and
+React Native, respectively). Generally speaking, this investment has turned out very well for us! Relay empowers
+product teams to quickly iterate on new features and to share common infrastructure across web and iOS codebases.
+However, most of the original engineers who pioneered using Relay at Artsy have since moved on to their next role;
+this has left a knowledge gap where Artsy engineers are comfortable _using_ Relay, but they don't totally
+understand how it works.
+
+This is a problem as old as software engineering itself, and it has a simple solution: learn and then teach others.
+We'll be driving a peer learning group centering around Relay, but today we are going to dive into the part of
+Relay that comes up the most in requests for pairing: getting Relay pagination to work.
+
+Let's back up and talk a little bit about what Relay is and how it works. Relay is a framework that glues React
+components and GraphQL requests together. React components define the data they need from a GraphQL schema in order
+to render themselves, and Relay handles actually fetching GraphQL requests and marshalling data into the React
+component tree. It is very efficient because of build-time optimizations by the Relay compiler.
+
+The simplest use of Relay is a [fragment container](https://relay.dev/docs/en/fragment-container), which is created
+from a React component and a [GraphQL fragment](https://blog.logrocket.com/graphql-fragments-explained/). (We're
+going to skip over how the GraphQL query is made, but
+[here are the docs on query renderers](https://relay.dev/docs/en/query-renderer) if you're curious.)
+
+```js
```

Comment:
```suggestion
```jsx
```

Reply:
I don't know that that matters actually, but maybe?

Reply:
Sadly, [pygment](https://pygments.org/languages/) doesn't yet support JSX: https://github.com/pygments/pygments/issues/709

Reply:
Ohh, gotcha. Didn't realize that!

Reply:
Based on my experience, the Pygments authors are very open to PR's for lexer support and bug fixes. If you're needing JSX support and have some Python chops you may be able to add support for this!